core/src/databases/table_schema.rs

```diff
@@ -262,6 +265,85 @@ impl TableSchema {
                 .collect::<Result<Vec<_>>>(),
         }
     }
+
+    pub fn merge(&self, other: &Self) -> Result<Self, anyhow::Error> {
+        // Index our and other columns by name.
+        let our_columns_set = self.0.iter().map(|c| &c.name).collect::<HashSet<_>>();
+        let other_columns_map = other
+            .0
+            .iter()
+            .map(|c| (&c.name, c))
+            .collect::<HashMap<_, _>>();
+
+        let merged_schema = self
+            .0
+            .iter()
+            // Iterate over all of our columns.
+            .map(|our_column| {
+                let mut our_column = our_column.clone();
+
+                // If the other schema has a column with the same name, merge it into our column.
+                if let Some(other_column) = other_columns_map.get(&our_column.name) {
+                    // If types are different, we need to merge them.
+                    if our_column.value_type != other_column.value_type {
+                        use TableSchemaFieldType::*;
+                        our_column.value_type =
+                            match (&our_column.value_type, &other_column.value_type) {
+                                // Ints and Floats can be merged into Floats.
+                                // Other types are widened to Text.
+                                (Int, Float) | (Float, Int) => TableSchemaFieldType::Float,
+                                _ => TableSchemaFieldType::Text,
```

Comment:
We can't go Text if we have boolean / int this would not be aligned with the json type?

Reply:
Do you prefer that we error ?

Reply:
Fixed. We accept widening for int/float but other than that it's error.